internal/devpkg/package.go

```diff
@@ -121,7 +121,7 @@ func PackageFromStringWithOptions(raw string, locker lock.Locker, opts devopt.Ad
 	pkg := PackageFromStringWithDefaults(raw, locker)
 	pkg.DisablePlugin = opts.DisablePlugin
 	pkg.PatchGlibc = opts.PatchGlibc
-	pkg.initOutputs(opts.Outputs)
+	pkg.outputs.selectedNames = lo.Uniq(append(pkg.outputs.selectedNames, opts.Outputs...))
```

Comment:
this feels brittle code for longer term, but okay for now

Reply:
It's a weird, but I thought it was simplest. It basically merges and dedups outputs that come from the flake and the outputs field in devbox.json packages. (which I think is reasonable?) the alternative would be an error no? 

I mostly didn't want to make a call on what to do if both are set. Currently flakes don't support the `--outputs` flag, but I think that's a bug/limitation we can fix.